### ISSUE TITLE
Mac OS Testing

### ISSUE THREAD

**Issue by dccsillag** [dccsillag/termimg#1]
@yingzhu146

I finally got around to starting to try out my image display ideas. It's extremely similar to what Ueberzug does, for now, but should be cross platform (if we don't count Windows, which is a problem for another day). Disclaimer: the code is still a bit hacky, it's just a prototype for now.

Could you try it out on Mac? Pick some image (one that easily fits in the screen) and do a `cargo run --release -- IMAGE_PATH ROW COL` where `IMAGE_PATH` is the path to the image, `ROW` is the terminal row in which to draw the image and `COL` is the terminal column in which to draw the image (both starting at 0); you can hit CtrlC to stop the display.

Does the image appear? It should appear on the current window, which should be the terminal you've just typed the command in. Is it properly aligned with the terminal's characters? Anything looking weird?

P.S.: The idea is to eventually use this as an image provider for NeoVim/magma-nvim, as a more portable alternative to Ueberzug.

**Comment by yingzhu146** [comment]
Ah so sorry I was watching the other issue not this. I'll try right now!

**Comment by yingzhu146** [comment]
Unfortunately, no!

```shell
/Users/yingzhu/termimg(main)> cargo run --release /Users/yinzghu/Downloads/qwe_download.jpeg 20 20
    Finished release [optimized] target(s) in 0.09s
     Running `target/aarch64-apple-darwin/release/termimg /Users/yinzghu/Downloads/qwe_download.jpeg 20 20`
Error: Couldn't connect to X

Caused by:
    Display parsing error
```

**Comment by dccsillag** [comment]
`echo $DISPLAY`?

**Comment by dccsillag** [comment]
(Meanwhile, I'm looking for alternatives, in case the current approach doesn't work in any way.)

**Comment by yingzhu146** [comment]
```shell
~ ❯ if [[ -z $DISPLAY ]]; then echo "empty"; else echo $DISPLAY; fi
empty
```

**Comment by yingzhu146** [comment]
not sure if relevant but:
using kitty, zsh, macOS Big Sur, arm64

```shell
~ ❯ rustup --version                                                                                                                      ✘ INT  base
rustup 1.24.3 (2021-05-31)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.56.0-nightly (574d37568 2021-08-07)`
~ ❯ cargo --version                                                                                                                              base
cargo 1.56.0-nightly (cc17afbb0 2021-08-02)

```

**Comment by dccsillag** [comment]
Try running `export DISPLAY=:0` and then cargo run? If that still doesn't work, then also run `xhost +` after the export?

**Comment by yingzhu146** [comment]
this is probably super obvious to you but do you expect `xhost` to be present that given that macOS doesn't run a X Window System?  There is XQuartz to work around it but it's very, very buggy in my experience, happy to set it up, to help you debug, although to be honest a solution that requires me to set up XQuartz, when Kitty/notcurses already solves the problem without requiring X11 server, seems not very attractive for MacOS users.

**Comment by dccsillag** [comment]
I though XQuartz already came built-in?

Well, if it doesn't, then using X11 indeed isn't a good path.
I've stumbled onto a possible _very_ cross platform solution, but will take a while to implement.

But I do agree that notcurses should be the focus magma-wise -- they already solve the problem for a lot of terminals. I hope to be able to work on it soon, but can't really promise any particular date for the time being.

**Comment by yingzhu146** [comment]
I don't think so :/ On my M1 Mac for sure not. And I've yet to see a single app work acceptably with XQuartz, I've always had rendering issues. But I'll try to set it up this weekend and give it a spin
